lib/blkproto.mli

```diff
@@ -0,0 +1,122 @@
+val max_segments_per_request : int
+
+module FeatureIndirect : sig
+  type t = { max_indirect_segments : int }
+
+  val _max_indirect_segments : string
+
+  val to_assoc_list : t -> (string * string) list
+end
+
+module State : sig
+  type t = | Initialising | InitWait | Initialised | Connected | Closing | Closed
+  val _state : string
+  val of_string : string -> t option
+  val to_assoc_list : t -> (string * string) list
+  val to_string : t -> string
+end
+
+module Mode : sig
+  type t = ReadOnly | ReadWrite
+end
+
+module Media : sig
+  type t = CDROM | Disk
+end
+
+module DiskInfo : sig
+  type t = {
+    sector_size : int;
+    sectors : int64;
+    media : Media.t;
+    mode : Mode.t;
+  }
+
+  val to_assoc_list : t -> (string * string) list
+end
+
+module Hotplug : sig
+  val _hotplug_status : string
+  val _online : string
+end
+
+module Connection : sig
+  type t = {
+    virtual_device: string;
+    backend_path: string;
+    backend_domid: int;
+    frontend_path: string;
+    frontend_domid: int;
+    mode: Mode.t;
+    media: Media.t;
+    removable: bool;
+  }
+
+  val to_assoc_list : t -> (int * (string * string)) list
+end
+
+module Req : sig
+  type seg = {
+    gref : int32;
```

Comment:
Maybe replace all `int32`s with `Gntref.t`, once #79 is merged?

Reply:
Yep, that's what I'd like to do, but I wanted #79 to be considered separately from this maintenance/quality-of-life stuff.